spot-ml/src/main/scala/org/apache/spot/dns/DNSSuspiciousConnectsAnalysis.scala

```diff
@@ -62,38 +62,45 @@ object DNSSuspiciousConnectsAnalysis {
     * @param logger
     */
   def run(config: SuspiciousConnectsConfig, sparkSession: SparkSession, logger: Logger,
-          inputDNSRecords: DataFrame): SuspiciousConnectsAnalysisResults = {
+          inputDNSRecords: DataFrame): Option[SuspiciousConnectsAnalysisResults] = {
 
 
     logger.info("Starting DNS suspicious connects analysis.")
 
-    val dnsRecords = filterRecords(inputDNSRecords)
-      .select(InSchema: _*)
-      .na.fill(DefaultQueryClass, Seq(QueryClass))
-      .na.fill(DefaultQueryType, Seq(QueryType))
-      .na.fill(DefaultQueryResponseCode, Seq(QueryResponseCode))
+    logger.info("Validating schema...")
+    val schemaValidationResults = validateSchema(inputDNSRecords)
 
-    logger.info("Fitting probabilistic model to data")
-    val model =
-      DNSSuspiciousConnectsModel.trainModel(sparkSession, logger, config, dnsRecords)
+    if (schemaValidationResults.length > InputSchema.ResponseDefaultSize) {
```

Comment:
So, the validator returns a Seq of messages (String). If everything went well, there will be 1 message (initializing message) but if there are schema errors there are going to be more than one. 
If is > InputSchema.ResponseDefaultSize (meaning 1) then it indicates there are some errors, if it's one then everything is good.